pkg/querier/blocks_store_queryable.go

```diff
@@ -209,11 +209,14 @@ func NewBlocksStoreQueryableFromConfig(querierCfg Config, gatewayCfg storegatewa
 	bucketClient = cachingBucket
 
 	// Create the blocks finder.
-	finder := NewBucketScanBlocksFinder(BucketScanBlocksFinderConfig{
-		ScanInterval:             storageCfg.BucketStore.SyncInterval,
-		TenantsConcurrency:       storageCfg.BucketStore.TenantSyncConcurrency,
-		MetasConcurrency:         storageCfg.BucketStore.MetaSyncConcurrency,
-		CacheDir:                 storageCfg.BucketStore.SyncDir,
+	finder := NewBucketIndexBlocksFinder(BucketIndexBlocksFinderConfig{
```

Comment:
NewBucketScanBlocksFinder looks unused now. Do you want to clean it up, along with tests and structs, etc?

Reply:
Good idea - will do!